File: src/main/java/net/slipp/web/smalltalk/SmallTalkContoller.java
package net.slipp.web.smalltalk;

import javax.annotation.Resource;

import net.slipp.domain.smalltalk.SmallTalk;
import net.slipp.domain.smalltalk.SmallTalkComment;
import net.slipp.domain.user.SocialUser;
import net.slipp.service.MailService;
import net.slipp.service.smalltalk.SmallTalkService;
import net.slipp.support.web.argumentresolver.LoginUser;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.validation.BindingResult;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

@Controller
public class SmallTalkContoller {

	private Logger log = LoggerFactory.getLogger(MailService.class);

	@Resource(name = "smallTalkService")
	private SmallTalkService smallTalkService;

	@RequestMapping(value = "/smalltalks", method = RequestMethod.POST)
	@ResponseBody
	public String save(@LoginUser SocialUser loginUser, @Validated SmallTalk smallTalk, BindingResult result) {
		try {
			if (result.hasErrors()) {
				throw new Exception();
			}
			smallTalk.setWriter(loginUser);
			smallTalkService.create(smallTalk);
		} catch (Exception e) {
			log.error("SmallTalk 데이터를 저장하는 중 오류. [Form] : {}", smallTalk, e);
			return "FAIL";
		}
		return "OK";
	}

	@RequestMapping(value = "/ajax/smalltalks", method = RequestMethod.GET)
	public String finds(Model model) {
		model.addAttribute("smallTalks", smallTalkService.getLastTalks());
		return "async/smalltalk";
	}
	
    @RequestMapping(value = "/ajax/smalltalks/{id}/comments", method = RequestMethod.POST)
    public @ResponseBody SmallTalkComment saveComment(@LoginUser SocialUser loginUser, @PathVariable Long id, SmallTalkComment smallTalkComment) {
        try {
            log.debug("Comments : {}", smallTalkComment);
            smallTalkComment.setWriter(loginUser);
            return smallTalkService.createComment(id, smallTalkComment);
        } catch (Exception e) {
            log.error("SmallTalkComment 데이터를 저장하는 중 오류. [Form] : {}", smallTalkComment, e);
            return new SmallTalkComment();
        }
    }
	
	@RequestMapping(value = "/ajax/smalltalks/{id}/comments", method = RequestMethod.GET)
	public String getComments(@PathVariable Long id, SmallTalkComment smallTalkComment, Model model) {
		try {
			log.debug("Comments : {}", smallTalkComment);
			model.addAttribute("smalltalkComments" ,smallTalkService.findCommnetsBySmallTalkId(id));
		} catch (Exception e) {
			log.error("SmallTalkComment 데이터를 저장하는 중 오류. [Form] : {}", smallTalkComment, e);
		}
		return "async/smalltalkcomments";
	}
}

File: src/main/scala/net/slipp/web/smalltalk/SmallTalkContoller.scala
package net.slipp.web.smalltalk

import javax.annotation.Resource
import javax.validation.Valid

import com.typesafe.scalalogging.LazyLogging
import net.slipp.domain.smalltalk.SmallTalkComment
import net.slipp.domain.user.SocialUser
import net.slipp.domain.smalltalk.SmallTalk
import net.slipp.service.smalltalk.SmallTalkService
import net.slipp.support.web.argumentresolver.LoginUser
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.{PathVariable, RequestMapping, RequestMethod, ResponseBody}

@Controller
class SmallTalkContoller(@Resource(name = "smallTalkService") smallTalkService: SmallTalkService) extends LazyLogging {
  @RequestMapping(value = Array("/smalltalks"), method = Array(RequestMethod.POST))
  @ResponseBody def save(@LoginUser loginUser: SocialUser, @Valid smallTalk: SmallTalk, result: BindingResult) = {
    if (result.hasErrors()) {
      throw new IllegalAccessException
    }

    try {
      smallTalk.setWriter(loginUser)
      smallTalkService.create(smallTalk)
      "OK"
    } catch {
      case e: Exception => {
        logger.error("SmallTalk 데이터를 저장하는 중 오류. [Form] : {}", smallTalk, e)
        "FAIL"
      }
    }
  }

  @RequestMapping(Array("/ajax/smalltalks"))
  def finds(model: Model) = {
    model.addAttribute("smallTalks", smallTalkService.getLastTalks)
    "async/smalltalk"
  }

  @RequestMapping(value = Array("/ajax/smalltalks/{id}/comments"), method = Array(RequestMethod.POST))
  @ResponseBody def saveComment(@LoginUser loginUser: SocialUser, @PathVariable id: Long, smallTalkComment: SmallTalkComment) = {
    try {
      logger.debug("Comments : {}", smallTalkComment)
      smallTalkComment.setWriter(loginUser)
      smallTalkService.createComment(id, smallTalkComment)
    } catch {
      case e: Exception => {
        logger.error("SmallTalkComment 데이터를 저장하는 중 오류. [Form] : {}", smallTalkComment, e)
        new SmallTalkComment()
      }
    }
  }

  @RequestMapping(Array("/ajax/smalltalks/{id}/comments"))
  def getComments(@PathVariable id: Long, smallTalkComment: SmallTalkComment, model: Model) = {
    try {
      logger.debug("Comments : {}", smallTalkComment)
      model.addAttribute("smalltalkComments", smallTalkService.findCommnetsBySmallTalkId(id))
      "async/smalltalkcomments"
    } catch {
      case e: Exception => {
        logger.error("SmallTalkComment 데이터를 저장하는 중 오류. [Form] : {}", smallTalkComment, e)
        "async/smalltalkcomments"
      }
    }
  }
  
  def this() = this(null)
}